### PR TITLE
docs: building container images with bazel (oci/wolfi) 

### PR DESCRIPTION
**Pull request by jhchabran**
Add first batch of documentation around build images with `rules_oci` and `wolfi`

## Test plan

local docs preview

### CODE REVIEW

File: doc/dev/background-information/bazel.md [hunk 1]
```diff
@@ -15,20 +15,19 @@ Sourcegraph is currently migrating to Bazel as its build system and this page is
...
-
-Other jobs will follow-up shortly, but as long as you're following instructions mentioned in that doc (TL;DR run `bazel configure`) this won't change anything for you.
-
-You may find the build and tests to be slow at first, either locally or in CI. This is because to be efficient, Bazel cache needs to be warm.
+- 2023-05-02 Additonally, to normal container building process, container images are now also built with `rules_oci` and Wolfi for as base images.
+  - Those new images are used by the new test targets under `//testing/...` that run the E2E tests and Backend Integration tests.
+  - The new images are deployed on S2 (and only S2).
+  - See [Slack thread](https://sourcegraph.slack.com/archives/C04MYFW01NV/p1685606796918859)
+- **2023-05-07 The new images are deployed on Dotcom.** 
+- 2023-05-14 After the release cut and the beginning of the code freeze, all operations that are now migrated to Bazel will have their legacy counterpart removed.
```
Comment (valerybugakov) [inline doc/dev/background-information/bazel.md:23]: :letsgoooo:

**Review by valerybugakov — APPROVED** [review]
Learned a ton of stuff from reading this! 💜

File: doc/dev/background-information/bazel_containers.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+- Sourcegraph Base Image 
+  - Distroless base, powered by Wolfi
+  - Packages common to all services. 
+- Per Service Base Image
+  - Packages specific to a given service 
+- Service specific outputs (`pkg_tar` rules) 
+  - Configuration files, if needed
+  - Binaries (targeting `linux/amd64`) 
+- Image configuration  (`oci_image` rules)
```
Comment (indradhanush) [inline doc/dev/background-information/bazel_containers.md:33]: `oci_image rules` is a new term for me. A link for follow up reading would be nice. Or is that bazel specific?

**Review by indradhanush — COMMENTED** [review]
Very nice intro doc! As someone not familiar with bazel apart from `bazel test //test/...` (😂) I found this doc very approachable. 👌 

Leaving approval for someone who is more familiar with this.

File: doc/dev/background-information/bazel_containers.md (hunk 2 above)
Comment (jhchabran) [reply doc/dev/background-information/bazel_containers.md:33]: `oci_image` is a _rule_ from the `oci_rules` package mentioned at the top of the doc. If that's the `oci` part that you're unfamiliar with, OCI stands for Open Container Interface. That's a good point, I'll drop a note about this at the top 🙏

**Pull request merged**